Content.Shared/GameObjects/Components/Sound/SharedSoundComponent.cs

```diff
@@ -0,0 +1,197 @@
+        /// </summary>
+        public virtual void StopScheduledSound(string filename)
+        {}
+
+        /// <summary>
+        /// Adds an scheduled sound to be played.
+        /// </summary>
+        public virtual void AddScheduledSound(ScheduledSound scheduledSound)
+        {}
+
+        /// <summary>
+        ///     Play an audio file globally, without position.
+        /// </summary>
+        /// <param name="filename">The resource path to the OGG Vorbis file to play.</param>
+        public void Play(string filename, AudioParams? audioParams = null)
+        {
+            AddScheduledSound(new ScheduledSound()
+            {
+                Filename = filename,
+                AudioParams = audioParams,
+                SoundType = SoundType.Global
+            });
+        }
+
+        /// <summary>
+        ///     Play an audio file following an entity.
+        /// </summary>
+        /// <param name="filename">The resource path to the OGG Vorbis file to play.</param>
+        /// <param name="entity">The entity "emitting" the audio.</param>
+        public void Play(string filename, IEntity entity, AudioParams? audioParams = null)
+        {
+            AddScheduledSound(new ScheduledSound()
+            {
+                Filename = filename,
+                AudioParams = audioParams,
+                SoundType = SoundType.Normal,
+                EntityUid = entity.Uid
+            });
+        }
+
+        /// <summary>
+        ///     Play an audio file at a static position.
+        /// </summary>
+        /// <param name="filename">The resource path to the OGG Vorbis file to play.</param>
+        /// <param name="coordinates">The coordinates at which to play the audio.</param>
+        public void Play(string filename, GridCoordinates coordinates, AudioParams? audioParams = null)
+        {
+            AddScheduledSound(new ScheduledSound()
+            {
+                Filename = filename,
+                AudioParams = audioParams,
+                SoundType = SoundType.Positional,
+                SoundPosition = coordinates
+            });
+        }
+    }
+
+    [NetSerializable, Serializable]
+    public class ScheduledSoundMessage : ComponentMessage
+    {
+        public ScheduledSound Schedule;
+        public ScheduledSoundMessage()
+        {
+            Directed = true;
+        }
+    }
+
+    [NetSerializable, Serializable]
+    public class StopSoundScheduleMessage : ComponentMessage
+    {
+        public string Filename;
+        public StopSoundScheduleMessage()
+        {
+            Directed = true;
+        }
+    }
+
+    [NetSerializable, Serializable]
+    public class StopAllSoundsMessage : ComponentMessage
+    {
+        public StopAllSoundsMessage()
+        {
+            Directed = true;
+        }
+    }
+
+    public enum SoundType
+    {
+        /// <summary>
+        /// Sound follows the entity.
+        /// </summary>
+        Normal,
```

Comment:
Rename this to `Entity` I'd say.

Reply:
Yeah, Entity seems more accurate than Normal. I'll change it.